scripts/synthetix/facilitate_trade.js

```diff
@@ -50,6 +51,30 @@ const gasStationURL = {
   4: "https://safe-relay.rinkeby.gnosis.io/api/v1/gas-station/",
 }
 
+const toWEI = function (value, decimals) {
+  const h = Math.floor(decimals / 2)
+  const a = new BN(10).pow(new BN(h))
+  const b = new BN(10).pow(new BN(decimals - h))
+
+  return new BN(value * a).mul(new BN(b))
+}
```

Comment:
Why not using `toErc20Units` here? (You'd have to convert float to string with no more decimals than `decimals` first, or better create a `floatToErc20Unit` that does that.)
I really don't like this function, I believe there are things that could go wrong here because of float shenanigans.
If `decimals=6` and `value=10/260`, then the output is different than expected by more than 1%. This is because the BN * float operation in line 59 truncates to the nearest integer.
I believe the reason why `h` exists is to juggle the amounts in `value*a` so that for common input values the result is an integer smaller than `2^52` but not so small that you lose too much precision in the truncation. It feels quite fragile.
Also, because of rounding issues I wouldn't be really confident that if `h = Math.floor(decimals / 2)` then `Math.floor(h) + Math.floor(decimals-h)` is the same integer as `decimals`, as this code requires.

Reply:
Yes, I would be happy to use our already existing functionality... I used this because I didn't realize we already had something. Will find it and make use of it asap!